content/docs/upgrading/upgrade.md

```diff
@@ -4,11 +4,105 @@ description = "Upgrading your deployment to a later version of Kubeflow"
 weight = 30
 +++
 
-Until version 1.0, Kubeflow makes no promises of backwards compatibility or 
-upgradeability. There is **no clean upgrade path** to the latest version of 
-Kubeflow ({{% kf-latest-version %}}).
+## Upgrading from Kubeflow v0.7.0 or later
+Upgrading your Kubeflow deployment is supported if your deployment is v0.7.0 or later.
 
-Nonetheless, here are some instructions for updating your deployment:
+### Prerequisites
+
+* Download the latest kfctl binary from the
+  [Kubeflow releases page](https://github.com/kubeflow/kubeflow/releases/tag/{{% kf-latest-version %}}).
+
+* Ensure that your Kubeflow namespace is annotated with the
+  `control-plane: kubeflow` label. You can verify this by doing:
+  ```
+  kubectl get namespace $NAMESPACE -o yaml > ns.yaml
+  ```
+
+    Examine the contents of the file. You should see something like:
+    ```
+    metadata:
+      labels:
+        control-plane: kubeflow
+        katib-metricscollector-injection: enabled
+    ```
+
+    If the label is not present, you can manually add it to the file, then do
+    ```
+    kubectl apply -f ns.yaml
+    ```
+    to update the existing namespace.
+
+* You must have a local Kubeflow application directory matching your current
+deployment. We'll call this `${KF_DIR}`.
+    * This should be present if you deployed Kubeflow using the kfctl tool.
+    * If you deployed Kubeflow using some other method (e.g. the 
+      [Deployment UI](/docs/gke/deploy/deploy-ui/)),
+      you will need to first create a local deployment directory.
+
+
+### Upgrade instructions
+
+1. Create an upgrade specification in the parent directory of your `${KF_DIR}`. For example, to upgrade
+a v0.7.0 deployment to v0.7.1, use
+[this specification](https://github.com/kubeflow/manifests/blob/v0.7-branch/kfdef/kfctl_upgrade_gcp_iap_0.7.1.yaml).
+Your directory structure should look like:
+    ```
+    ${PARENT_DIR}
+    |----${KF_DIR}
+    |----kfctl_upgrade_spec.yaml
+
+    ```
+
+1. Modify the configuration for your deployment:
+    ```
+    apiVersion: kfupgrade.apps.kubeflow.org/v1alpha1
+    kind: KfUpgrade
+    metadata:
+      name: kf-upgrade-v0.7.1
+    spec:
+      currentKfDef:
+        # Replace with the name of your Kubeflow app
+        name: kubeflow-app
+        version: v0.7.0
+      newKfDef:
+        # Replace with the name of your kubeflow app
+        name: kubeflow-app
+        version: v0.7.1
+      # Replace this with the path to the KfDef that you are upgrading to
+      baseConfigPath: https://raw.githubusercontent.com/kubeflow/manifests/v0.7-branch/kfdef/kfctl_gcp_iap.0.7.1.yaml
+    ```
+  
+1. Run the `apply` command to upgrade your deployment:
```

Comment:
How about the code below to fix the formatting problem. **Note:** Please view this comment in edit mode to see the code - I couldn't put it into a code block because it contains code blocks.

======= CODE IS BELOW THIS LINE ====

If you don't need to change any kustomize parameter values in your configuration, run the `apply` command now to upgrade your deployment:

    ```
    kfctl apply -f ${UPGRADE_SPEC} -V
    ```

Alternatively you can follow these steps to change the configuration before applying the upgrade:

1. Run a `build` command:

    ```
    kfctl build -f ${UPGRADE_SPEC} -V
    ```

      The above command creates a new Kubeflow application in the same directory, with a name
      in the form of a 7-character long hash value. The directory structure should look like this:

    ```
    ${PARENT_DIR}
    |----${KF_DIR}
    |----${UPGRADE_DIR}
    |----kfctl_upgrade_spec.yaml
    ```

      You can examine and change the kustomize parameter values in `${UPGRADE_DIR}`.

1. Run the `apply` command to upgrade the deployment:

     ```
    kfctl apply -f ${UPGRADE_SPEC} -V
    ```